src/v0/destinations/ga4/customMappingsHandler.js

```diff
@@ -0,0 +1,198 @@
+const get = require('get-value');
+const jsonpath = require('rs-jsonpath');
+const {
+  validateEventName,
+  basicValidation,
+  isReservedEventName,
+  addClientDetails,
+  removeReservedParameterPrefixNames,
+  prepareUserConsents,
+  removeInvalidParams,
+  GA4_RESERVED_PARAMETER_EXCLUSION,
+  getGA4CustomParameters,
+  buildDeliverablePayload,
+  GA4_PARAMETERS_EXCLUSION,
+  prepareUserProperties,
+} = require('./utils');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const {
+  removeUndefinedAndNullRecurse,
+  constructPayload,
+  isDefinedAndNotNull,
+  isEmptyObject,
+  removeUndefinedAndNullValues,
+  isHybridModeEnabled,
+  getIntegrationsObj,
+} = require('../../util');
+const { trackCommonConfig, ConfigCategory, mappingConfig } = require('./config');
+
+const findGA4Events = (eventsMapping, event) => {
+  // Find the event using destructuring and early return
+  const validMappings = eventsMapping.filter(
+    (mapping) =>
+      mapping.rsEventName?.trim().toLowerCase() === event.trim().toLowerCase() &&
+      mapping.ga4EventName,
+  );
+  // Return an empty object if event not found
+  return validMappings;
+};
+
+const handleCustomMappings = (message, Config) => {
+  const { eventsMapping } = Config;
+
+  let rsEvent = get(message, 'event');
+  basicValidation(rsEvent);
+
+  const validMappings = findGA4Events(eventsMapping, rsEvent);
+
+  if (validMappings.length === 0) {
+    // Default mapping
+
+    let rawPayload = constructPayload(message, trackCommonConfig);
+
+    const ga4EventPayload = {};
+
+    ga4EventPayload.name = rsEvent;
+
```

Comment:
Before assigning event name from `Config.eventMappings`, validation should be done so that reserved names are not being entered by the user - https://github.com/rudderlabs/rudder-transformer/pull/3289/files#diff-343714a0fc63a78f89f6c19fce1483e1c57107605c9b7cebcea0595f977b7b93R239

Reply:
This is happening on Line 99

Reply:
That's for events that have mappings but shouldn't that happen for default flow too ?

Reply:
Addressed